rotkehlchen/chain/ethereum/modules/ammswap/ammswap.py

```diff
@@ -0,0 +1,561 @@
+            )
+            for _, trades in trade_mapping.items():
+                all_trades.extend(trades)
+
+            return all_trades
+
+    def _fetch_trades_from_db(
+            self,
+            addresses: List[ChecksumEthAddress],
+            from_timestamp: Timestamp,
+            to_timestamp: Timestamp,
+    ) -> AddressTrades:
+        """Fetch all DB Sushiswap trades within the time range"""
+        try:
+            location = self.location  # type: ignore
+        except AttributeError as e:
+            raise ModuleInitializationFailure('Location not properly set') from e
+
+        db_address_trades: AddressTrades = {}
+        for address in addresses:
+            db_swaps = self.database.get_amm_swaps(
+                from_ts=from_timestamp,
+                to_ts=to_timestamp,
+                location=location,
+                address=address,
+            )
+            db_trades = self.swaps_to_trades(db_swaps)
+            if db_trades:
+                db_address_trades[address] = db_trades
+
+        return db_address_trades
+
+    @abc.abstractmethod
+    def _get_trades_graph_for_address(
+            self,
+            address: ChecksumEthAddress,
+            start_ts: Timestamp,
+            end_ts: Timestamp,
+    ) -> List[AMMTrade]:
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def get_trades_history(
+        self,
+        addresses: List[ChecksumEthAddress],
+        reset_db_data: bool,
+        from_timestamp: Timestamp,
+        to_timestamp: Timestamp,
+    ) -> AddressTrades:
+        """Get the addresses' trades history in the AMMswap protocol"""
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def _get_trades(
+            self,
+            addresses: List[ChecksumEthAddress],
+            from_timestamp: Timestamp,
+            to_timestamp: Timestamp,
+            only_cache: bool,
+    ) -> AddressTrades:
+        """Request via graph all trades for new addresses and the latest ones
+        for already existing addresses. Then the requested trade are written in
+        DB and finally all DB trades are read and returned.
+        """
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def _get_events_balances(
+            self,
+            addresses: List[ChecksumEthAddress],
+            from_timestamp: Timestamp,
+            to_timestamp: Timestamp,
+    ) -> AddressEventsBalances:
+        """Request via graph all events for new addresses and the latest ones
+        for already existing addresses. Then the requested events are written
+        in DB and finally all DB events are read, and processed for calculating
+        total profit/loss per LP (stored within <LiquidityPoolEventsBalance>).
+        """
+        raise NotImplementedError
+
+    # -- Methods following the EthereumModule interface -- #
+
+    @abc.abstractmethod
+    def deactivate(self) -> None:
+        raise NotImplementedError
```

Comment:
```suggestion
    # -- Methods following the EthereumModule interface -- #

    @abc.abstractmethod
    def deactivate(self) -> None:
        raise NotImplementedError
```

Why is this needed? This is not a module. And this will conflict with the implementation of deactivate() in the other EthereumModule supercass since you do multiple inheritance. Don't do that.

Reply:
It was because the other interface methods were the same and I wanted to get them here but it was not okey for the linter. And this one I didn't notice but is true that is the same in both and get overwritten. Going to remove it from here